src/main/java/com/stuypulse/robot/commands/ShooterControlCommand.java

```diff
@@ -0,0 +1,71 @@
+package com.stuypulse.robot.commands;
+
+import com.stuypulse.robot.Constants;
+import com.stuypulse.robot.subsystems.Shooter;
+import com.stuypulse.stuylib.input.WPIGamepad;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class ShooterControlCommand extends CommandBase {
+    public Shooter shooter;
+    public WPIGamepad gamepad;
+    public double targetVelocity;
+
+    public ShooterControlCommand(Shooter shooter, WPIGamepad gamepad) {
+        this.shooter = shooter;
+        this.gamepad = gamepad;
+        addRequirements(shooter);
+    }
+
+    @Override
+    public void execute() {
+        // If up dpad button is pressed, shoot from far (Steph Curry)
+        if (gamepad.getRawDPadUp()) {
+            targetVelocity = Constants.SHOOT_FROM_FAR_RPM;
+
+            shooter.retractHoodSolenoid();  
+            shooter.runShooter();
+            shooter.runFeeder();
+            
+            if (shooter.getCurrentShooterVelocityInRPM() > targetVelocity) {
+                gamepad.setRumble(1);
+            }
+        }
+        
+        // If down dpad button is pressed, shoot from initiation line
+        if (gamepad.getRawDPadDown()) {
+            targetVelocity = Constants.SHOOT_FROM_INITATION_LINE_RPM;
+
+            shooter.retractHoodSolenoid();
+            shooter.runShooter();
+            shooter.runFeeder();
+
+            if (shooter.getCurrentShooterVelocityInRPM() > targetVelocity) {
```

Comment:
Just because it is above the target velocity doesn’t mean that it is good, you should make a threshold with an upper and lower bound